cs/src/core/Transcoder.cs

```diff
@@ -61,15 +63,20 @@ static class Cache<R, W>
     /// <typeparam name="W">Protocol writer</typeparam>
     public class Transcoder<R, W>
     {
-        readonly Action<R, W>[] transcode;
-
         /// <summary>
         /// Create a transcoder for payloads with specified runtime schema
         /// </summary>
         /// <param name="schema">Payload schema, required for transcoding from untagged protocols</param>
         public Transcoder(RuntimeSchema schema)
         {
-            transcode = Generate(schema);
+            if (typeof(W).GetAttribute<FirstPassWriterAttribute>() != null)
+            {
+                helper = new TwoPassTranscoderHelper<CompactBinaryCounter>(schema);
```

Comment:
Shouldn't the type here depend on the value of `FirstPassWriterAttribute.Type` instead of being hard-coded to `CompactBinaryCounter`?